src/v/cluster/health_monitor_frontend.cc

```diff
@@ -120,7 +120,11 @@ ss::future<> health_monitor_frontend::update_disk_health_cache() {
           "Update disk health cache {} -> {}",
           _cluster_disk_health,
           disk_health);
-        _cluster_disk_health = disk_health;
+        // Two threads write to _cluster_disk_health async: (1) the timer
+        // callback via disk_health_tick and (2) refresh_info. So protect that
+        // state with a lock
+        co_await _disk_health_lock.with(
+          [this, disk_health]() { _cluster_disk_health = disk_health; });
```

Comment:
why do you need a lock here? it seems to be protecting `_cluster_disk_health = disk_health;` but that assignment doesn't contain any scheduling points.

Reply:
From what I could tell, there are two fibers in the health_monitor_frontend:
1. A background fiber in `disk_health_tick` which calls `update_frontend_and_backend_cache` and leads to `_cluster_disk_health = disk_health`
2. The fiber where `reset_info` runs which calls `update_frontend_and_backend_cache` and leads to `_cluster_disk_health = disk_health`

Therefore, I am concerned about concurrent writes to `_cluster_disk_health`